bpf/lib/nat.h

```diff
@@ -393,25 +390,37 @@ snat_v4_rewrite_headers(struct __ctx_buff *ctx, __u8 nexthdr, int l3_off,
 				return DROP_CSUM_L4;
 #endif  /* ENABLE_SCTP */
 			case IPPROTO_ICMP:
-				/* Not initialized by csum_l4_offset_and_flags(): */
+				/* Not initialized by csum_l4_offset_and_flags(), because ICMPv4
+				 * doesn't use a pseudo-header, and the change in IP addresses is
+				 * not supposed to change the L4 checksum.
+				 * Set it temporarily to amend the checksum after changing ports.
+				 */
 				csum.offset = offsetof(struct icmphdr, checksum);
-				/* No Pseudo-Hdr checksum for ICMPv4: */
-				flags = 0;
 				break;
 			default:
 				return DROP_UNKNOWN_L4;
 			}
 
-			l4_sum = csum_diff(&from, 4, &to, 4, 0);
-			if (l4_store_port(ctx, l4_off, port_off, new_port) < 0)
-				return DROP_WRITE_ERROR;
+			/* Amend the L4 checksum due to changing the ports. The L3 checksum stays
+			 * the same because changing both the port and the L4 checksum compensate
```

Comment:
This I don't quite parse - the IPv4 header checksum shouldn't even cover the L4 header (or any change to its port fields) ?